internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -19,6 +19,7 @@ package clusterclass
 import (
```

Comment:
I assume unit tests for the new additional behavior are TODO
(I'll assume the same in case I see other untested code in other files)

Reply:
I think this is the only place where there should be untested code in this PR - if you spot somewhere else let me know.

Reply:
Will do.

just fyi. Something else came up, will continue the review tomorrow